internal/providers/util/unzip.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2024 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"bytes"
+	"compress/gzip"
+	"io"
+)
+
+var (
+	gzipMagic = []byte{0x1F, 0x8B, 0x08}
+)
+
+func hasGzipMagicNumber(source []byte) bool {
+	return bytes.HasPrefix(source, gzipMagic)
+}
+
+func GunzipIfNeeded(raw []byte) ([]byte, error) {
+	if hasGzipMagicNumber(raw) {
```

Comment:
What do you think about inverting this logic so we can un-nest the bulk of the logic here ?
```
if !hasGzipMagicNumber(raw) { return raw, nil }
...
```